docs/internal/GeneralArchitectureGuide.md

```diff
@@ -6,6 +6,60 @@
 
 ## Settings
 
+Elasticsearch supports [node-level configuration file settings][], [cluster-level settings][] and [index-level settings][].
+
+### Declaring a Setting
+
+[node-level configuration file settings]: https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html
+[cluster-level settings]: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-update-settings.html
+[index-level settings]: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-update-settings.html
+
+The [Setting][] class is the building block for Elasticsearch server settings. Each `Setting` can take multiple [Property][]
+declarations to define setting characteristics, like allowing [dynamic updates][] while the node is running, or the scope of
+the setting to apply at the individual [index-level][] vs [cluster-level][]. A [dynamic Setting][] is often paired with a
+[local volatile variable][] and [ClusterSettings#initializeAndWatch()][] to catch and immediately apply dynamic changes.
```

Comment:
Probably worth observing here that (a) if a dynamic setting is not set then its value comes from the node-local setting as defined in `elasticsearch.yml`, only falling all the way back to the setting default if has no node-local value either, and (b) `initializeAndWatch` is the right way to achieve this fall-back behaviour. If you get a setting value directly from `org.elasticsearch.cluster.metadata.Metadata#settings` then you will see dynamic values and default values but not node-local values, which would be a bug. We've had bugs like that in the past.